spec/src/main/asciidoc/core/packagingdeployment.asciidoc

```diff
@@ -102,93 +94,11 @@ The container automatically discovers managed beans (according to the rules of <
-* the package name of the type being discovered starts with the value of the name attribute with a suffix ".**" of the exclude filter
-
-then we say that the type is excluded from discovery.
-
-For example, consider the follow `beans.xml` file:
-
-[source, xml]
-----
-<?xml version="1.0" encoding="UTF-8"?>
-<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
-       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-       xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/beans_3_0.xsd"
-       bean-discovery-mode="all" version="3.0">
-
-    <scan>
-        <exclude name="com.acme.rest.*" />
-
-        <exclude name="com.acme.faces.**">
-            <if-class-not-available name="jakarta.faces.context.FacesContext"/>
-        </exclude>
-
-        <exclude name="com.acme.verbose.*">
-            <if-system-property name="verbosity" value="low"/>
-        </exclude>
-
-        <exclude name="com.acme.ejb.**">
-            <if-class-available name="jakarta.enterprise.inject.Model"/>
-            <if-system-property name="exclude-ejbs"/>
-        </exclude>
-    </scan>
-
-</beans>
-----
-
-The first exclude filter will exclude all classes in `com.acme.rest` package. The second exclude filter will exclude all classes in the `com.acme.faces` package, and any subpackages, but only if JSF is not available. The third exclude filter will exclude all classes in the `com.acme.verbose` package if the system property `verbosity` has the value `low`. The fourth exclude filter will exclude all classes in the `com.acme.ejb` package, and any subpackages if the system property `exclude-ejbs` is set (with any value) and at the same time, the `jakarta.enterprise.inject.Model` class is available to the classloader.
-
-
-[[trimmed_bean_archive]]
-
-==== Trimmed bean archive
-
-An explicit bean archive may be marked as 'trimmed' by adding the `<trim />` element to its `beans.xml` file:
-
-[source, xml]
-----
-<?xml version="1.0" encoding="UTF-8"?>
-<beans xmlns="https://jakarta.ee/xml/ns/jakartaee"
-        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-        xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/beans_3_0.xsd"
-        version="3.0">
-
-    <trim/>
-</beans>
-----
-
-If an explicit bean archive contains the `<trim/>` element in its `beans.xml` file, types that don't have either a bean defining annotation (as defined in <<bean_defining_annotations>>) or any scope annotation, are removed from the set of discovered types.
+The container discovers each Java class with a bean defining annotation in an implicit bean archive.
 
+The container must also execute `@Discovery` extension methods on build compatible extensions and discover all classes added using the `ScannedClasses` API.
 
+When all types are discovered, the container must execute `@Enhancement` extension methods on build compatible extensions and alter its metadata representation of discovered types.
```

Comment:
```suggestion
When all types are discovered, the container must execute `@Enhancement` extension methods in build compatible extensions and alter metadata representation of discovered types accordingly.
```

Reply:
I added "accordingly". I don't think I agree with "on" -> "in".